### PR TITLE
Created infra for WLS Logging Exporter and converted related tests to use JUnit5

### PR DESCRIPTION
**Pull request by hzhao-github**
Use case
-------------
1. Created unit methods in Junit 5 test framework to install WebLogic Logging Exporter 
2. Install Kibana/Elasticsearch, install and start Operator with ELK Stack enabled and install WebLogic logging exporter in all WebLogic server pods
3. Create and start the WebLogic domain
4. Test WebLogic Logging Exporter by verify that Using WebLogic logging exporter,
     1) WebLogic server Logs is integrated to ELK Stack in the same pod that the domain is running on
     2) Log hits for WebLogic servers are not empty
     3) Filters is applied to all the messages

Jenkins Job passed:

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1516/parameters/

Jenkins job passed again with latest change

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1542/

Jenkins job passed again

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1558/

latest Jenkins job, one failure is not related to this change

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/1685/

### CODE REVIEW

**Comment by anpanigr** [comment]
Remove ItElasticLogging.java from Junit4 Directory in the current PR, if you covered the usecase

**Comment by hzhao-github** [comment]
Not ready to remove yet, Pani, testWebLogicLogSearch() is not integrated due to the issue described in JIRA OWLS-84057

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 1]
```diff
@@ -102,8 +121,14 @@
   private static LoggingExporterParams kibanaParams = null;
   private static LoggingFacade logger = null;
 
+  private static String k8sExecCmdPrefix;
+  private static Map<String, String> testVarMap;
+  private static final int maxIterationsPod = 10;
+
   /**
-   * Install Elasticsearch, Kibana and Operator and verify, create a one cluster domain.
+   * Install Elasticsearch, Kibana and Operator and verify,
```
Comment (anpanigr) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:129]: Can u re-phrase the description

Install Elasticsearch, Kibana and Operator.
Install WebLogic logging exporter in all WebLogic server pods to collect WebLogic logs.

[create a one cluster domain.  does not make sense  ]

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 2]
```diff
@@ -70,28 +78,39 @@
...
- * 2. Install and start Operator with ELK Stack enabled
+ * 1. Install Kibana/Elasticsearch.
+ * 2. Install and start Operator with ELK Stack enabled.
  * 3. Verify that ELK Stack is ready to use by checking the index status of
  *    Kibana and Logstash created in the Operator pod successfully
- * 4. Create and start the WebLogic domain
- * 5. Verify that Elasticsearch collects data from WebLogic logs and
- *    stores them in its repository correctly.
+ * 4. Install WebLogic logging exporter in all WebLogic server pods by
+ *    adding it to MII via --additionalBuildCommands and --additionalBuildFiles.
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:86]: changed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 3]
```diff
@@ -70,28 +78,39 @@
 
 /**
  * To test ELK Stack used in Operator env, this Elasticsearch test does
- * 1. Install Kibana/Elasticsearch 
- * 2. Install and start Operator with ELK Stack enabled
+ * 1. Install Kibana/Elasticsearch.
+ * 2. Install and start Operator with ELK Stack enabled.
  * 3. Verify that ELK Stack is ready to use by checking the index status of
  *    Kibana and Logstash created in the Operator pod successfully
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:84]: added

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 4]
```diff
@@ -216,14 +249,84 @@ public void testOperatorLogSearch() {
...
+
+    // Verify that the ms status of "RUNNING" is found in query return from Elasticsearch repos
+    verifyServerRunningInSearchResults(managedServerPodPrefix + "1");
+
+    logger.info("Query Operator log for WebLogic server status info succeeded");
+  }
+
+  /**
+   * Use Elasticsearch Search APIs to query WebLogic log info pushed to Elasticsearch repository
+   * by WebLogic logging exporter . Verify that log hits for WebLogic servers are not empty
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:276]: changed

File: new-integration-tests/src/test/resources/loggingexporter/copy-logging-files-cmds.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[final-build-commands]
+
+RUN mkdir -p /u01/domains/elk-domain1/config && chown oracle:oracle /u01/domains/elk-domain1/config
+COPY --chown=oracle:oracle files/WebLogicLoggingExporter.yaml /u01/domains/elk-domain1/config/
+
+RUN mkdir -p /u01/domains/elk-domain1/lib && chown oracle:oracle /u01/domains/elk-domain1/lib
```
[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/resources/loggingexporter/copy-logging-files-cmds.txt:6]: we have this util to replace values in the file:
FileUtils.replaceStringInFile(out, entry.getKey(), entry.getValue());

[the remark this answers is not in the document]
Comment (marinakog) [reply new-integration-tests/src/test/resources/loggingexporter/copy-logging-files-cmds.txt:6]: /**
   * Replaces each substring in the file that matches the given regular
   * expression with the given replacement.
   * @param filePath file in which a string has to be replaced
   * @param regex the regular expression to which this string is to be matched
   * @param replacement the string to be substituted for each match
   * @throws IOException if an IO error occurs while reading from the file
   */
  public static void replaceStringInFile(String filePath, String regex, String replacement)

[the remark this answers is not in the document]
Comment (hzhao-github) [reply new-integration-tests/src/test/resources/loggingexporter/copy-logging-files-cmds.txt:6]: Changed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/LoggingExporter.java [hunk 6]
```diff
@@ -468,4 +562,69 @@ private static String execLoggingExpStatusCheck(String namespace, String labelSe
...
+        .append(" wget -P ")
+        .append(wlsLoggingExporterArchiveLoc)
+        .append(" --server-response --waitretry=5 --retry-connrefused ")
+        .append(jarReposUrl)
+        .append("/")
+        .append(jarFileName);
+    logger.info("Executing cmd " + getJars.toString());
+
+    // Make sure downloading completed
+    while (i < maxIterationsPod) {
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/LoggingExporter.java:587]: variable name maxIterationsPod is not appropriate here, also use awaitility api like we do in other cases ...

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 7]
```diff
@@ -200,30 +235,100 @@ public void testLogLevelSearch() {
...
+  public void testWlsLoggingExporter() throws Exception {
+    Map<String, String> wlsMap = verifyLoggingExporterReady(opNamespace, null, WEBLOGIC_INDEX_KEY);
+    // merge testVarMap and wlsMap
+    testVarMap.putAll(wlsMap);
+
+    // Verify that occurrence of log level = Notice are not empty
+    String regex = ".*took\":(\\d+),.*hits\":\\{(.+)\\}";
+    String queryCriteria = "/_search?q=level:Notice";
+    verifyCountsHitsInSearchResults(queryCriteria, regex, WEBLOGIC_INDEX_KEY, false);
+    // Verify that occurrence of loggerName = WebLogicServer are not empty
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:291]: leave a blank line before this line for better code readability

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 8]
```diff
@@ -200,30 +235,100 @@ public void testLogLevelSearch() {
...
+    queryCriteria = "/_search?q=loggerName:WebLogicServer";
+    verifyCountsHitsInSearchResults(queryCriteria, regex, WEBLOGIC_INDEX_KEY, false);
+    // Verify that occurrence of _type:doc are not empty
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:294]: same as above

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 9]
```diff
@@ -200,30 +235,100 @@ public void testLogLevelSearch() {
...
+    queryCriteria = "/_search?q=_type:doc";
+    verifyCountsHitsInSearchResults(queryCriteria, regex, WEBLOGIC_INDEX_KEY, false);
+    // Verify that serverName:managed-server1 is filtered out
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:297]: same as above

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java [hunk 10]
```diff
@@ -329,7 +438,33 @@ private static void createDomainCrAndVerify(String adminSecretName,
     createDomainAndVerify(domain, domainNamespace);
   }
 
-  private void verifySearchResults(String queryCriteria, String regex,
+  private void verifyServerRunningInSearchResults(String serverName) {
+    int i = 0;
+    String queryResult = null;
+    String queryCriteria = "/_search?q=log:" + serverName;
+    while (i < maxIterationsPod) {
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:445]: variable name maxIterationsPod is not appropriate here, also use awaitility api like we do in other cases ...

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 7 above)
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:291]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 8 above)
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:294]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 9 above)
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:297]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/LoggingExporter.java (hunk 6 above)
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/LoggingExporter.java:587]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java (hunk 10 above)
Comment (hzhao-github) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLogging.java:445]: changed

**Pull request merged**